### ISSUE TITLE
Cannot save configuration files with sleap gui

### ISSUE THREAD

**Issue by bdy2530** [talmolab/sleap#1251]
# Bug description
I cannot save configuration files via the sleap gui. When I press "Save configuration files..." after making some changes, no dialogue window pops up. 

## Expected behaviour

A Finder window should pop up when I press "Save configuration files...", where I can choose the save location

## Actual behaviour

Nothing happens after pressing "Save configuration files..."

## Your personal set up
MacBook Pro 2021, M1 Pro, installed via the Apple Silicon Macs method

- OS: macOS 13.3

- Version(s): SLEAP v1.3.0, python 3.9.16

- SLEAP installation method:
  - [X] [Apple Silicon Macs](https://sleap.ai/installation.html#apple-silicon-macs)

<details><summary>Environment packages</summary>

```
(base) baran@dhcp-10-102-42-204 ~ % conda activate sleap
(sleap) baran@dhcp-10-102-42-204 ~ % pip freeze
absl-py==1.4.0
accessible-pygments==0.0.4
alabaster==0.7.13
anyio==3.6.2
appdirs==1.4.4
appnope==0.1.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asttokens==2.2.1
astunparse==1.6.3
attrs==21.4.0
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.0
black==21.6b0
bleach==6.0.0
cached-property @ file:///home/conda/feedstock_root/build_artifacts/cached_property_1615209429212/work
cachetools==5.3.0
cattrs==1.1.1
certifi==2021.10.8
cffi==1.15.1
cfgv==3.3.1
charset-normalizer==3.1.0
click==8.0.4
colorama==0.4.6
comm==0.1.3
commonmark==0.9.1
contourpy==1.0.7
coverage==7.2.2
cryptography==40.0.1
cycler==0.11.0
debugpy==1.6.6
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.13
distlib==0.3.6
docutils==0.17.1
efficientnet==1.0.0
entrypoints==0.4
exceptiongroup==1.1.1
executing==1.2.0
filelock==3.10.7
flatbuffers==1.12
fonttools==4.39.2
gast==0.4.0
gitdb==4.0.10
GitPython==3.1.31
google-auth==2.16.3
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio @ file:///Users/runner/miniforge3/conda-bld/grpc-split_1662408182534/work
h5py @ file:///Users/runner/miniforge3/conda-bld/h5py_1637964045648/work
hdmf==3.5.2
identify==2.5.22
idna==3.4
image-classifiers==1.0.0
imageio==2.15.0
imagesize==1.4.1
imgaug==0.4.0
imgstore==0.2.9
importlib-metadata==6.1.0
importlib-resources==5.12.0
iniconfig==2.0.0
ipykernel==6.22.0
ipython==8.11.0
ipython-genutils==0.2.0
ipywidgets==7.7.4
jaraco.classes==3.2.3
jedi==0.17.2
Jinja2==3.1.2
joblib==1.2.0
jsmin==3.0.1
json5==0.9.11
jsonpickle==1.2
jsonschema==4.17.3
jupyter-cache==0.4.3
jupyter-server==1.13.5
jupyter-server-mathjax==0.2.6
jupyter-sphinx==0.3.2
jupyter_client==8.1.0
jupyter_core==5.3.0
jupyterlab==3.3.4
jupyterlab-pygments==0.2.2
jupyterlab-widgets==1.1.3
jupyterlab_server==2.16.3
keras==2.9.0
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.2
keyring==23.13.1
kiwisolver==1.4.4
libclang==16.0.0
linkify-it-py==2.0.0
livereload==2.6.3
lxml==4.9.2
Markdown==3.4.3
markdown-it-py==1.1.0
MarkupSafe==2.1.2
matplotlib==3.7.1
matplotlib-inline==0.1.6
mdit-py-plugins==0.2.8
mistune==0.8.4
more-itertools==9.1.0
mypy-extensions==1.0.0
myst-nb==0.13.2
myst-parser==0.15.2
nbclassic==0.5.3
nbclient==0.5.13
nbconvert==6.5.4
nbdime==3.1.1
nbformat==5.1.3
ndx-pose==0.1.1
nest-asyncio==1.5.6
networkx==3.0
nixio==1.5.3
nodeenv==1.7.0
notebook==6.5.3
notebook_shim==0.2.2
numpy @ file:///Users/runner/miniforge3/conda-bld/numpy_1651020584890/work
oauthlib==3.2.2
opencv-python==4.6.0
opt-einsum==3.3.0
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1673482170163/work
pandas==1.5.3
pandocfilters==1.5.0
parso==0.7.1
pathspec==0.11.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==8.4.0
pkginfo==1.9.6
platformdirs==3.2.0
pluggy==1.0.0
pre-commit==3.2.1
prometheus-client==0.16.0
prompt-toolkit==3.0.38
protobuf==3.19.6
psutil==5.9.4
ptyprocess==0.7.0
pure-eval==0.2.2
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
pydata-sphinx-theme==0.13.1
PyGithub==1.58.1
Pygments==2.14.0
PyJWT==2.6.0
pykalman==0.9.5
PyNaCl==1.5.0
pynwb==2.3.1
pyparsing==3.0.9
pyrsistent==0.19.3
PySide2==5.15.5
pytest==7.2.2
pytest-cov==3.0.0
pytest-qt==4.2.0
pytest-xvfb==2.0.0
python-dateutil==2.8.2
python-rapidjson==1.10
pytz==2023.2
pytz-deprecation-shim==0.1.0.post0
PyVirtualDisplay==3.0
PyWavelets==1.4.1
PyYAML==6.0
pyzmq==25.0.2
qimage2ndarray==1.10.0
QtPy @ file:///home/conda/feedstock_root/build_artifacts/qtpy_1667873092748/work
readme-renderer==37.3
regex==2023.3.23
requests==2.28.2
requests-oauthlib==1.3.1
requests-toolbelt==0.10.1
rfc3986==2.0.0
rich==10.16.1
rsa==4.9
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.7
scikit-image==0.19.3
scikit-learn==1.0.2
scikit-video==1.1.11
SciPy @ file:///Users/runner/miniforge3/conda-bld/scipy_1667960661844/work
seaborn==0.12.2
segmentation-models==1.0.1
Send2Trash==1.8.0
Shapely @ file:///Users/runner/miniforge3/conda-bld/shapely_1622660864412/work
shiboken2==5.15.5
six==1.15.0
-e git+https://github.com/talmolab/sleap.git@b305eda5b1fc19955bb4e6014d0707b2938d0b62#egg=sleap
smmap==5.0.0
sniffio==1.3.0
snowballstemmer==2.2.0
soupsieve==2.4
Sphinx==4.5.0
sphinx-autobuild==2021.3.14
sphinx-book-theme==1.0.0
sphinx-copybutton==0.5.1
sphinx-togglebutton==0.3.2
sphinxcontrib-applehelp==1.0.4
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
SQLAlchemy==1.4.47
stack-data==0.6.2
tensorboard==2.9.1
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow-estimator==2.9.0
tensorflow-hub==0.13.0
tensorflow-macos==2.9.2
tensorflow-metal==0.5.0
termcolor==2.2.0
terminado==0.17.1
threadpoolctl==3.1.0
tifffile==2023.3.21
tinycss2==1.2.1
toml==0.10.2
tomli==2.0.1
tornado==6.2
tqdm==4.65.0
traitlets==5.9.0
twine==3.3.0
typing_extensions==4.5.0
tzdata==2023.2
tzlocal==4.3
uc-micro-py==1.0.1
urllib3==1.26.15
virtualenv==20.21.0
wcwidth==0.2.6
webencodings==0.5.1
websocket-client==1.5.1
Werkzeug==2.2.3
widgetsnbextension==3.6.3
wrapt==1.15.0
zipp==3.15.0
```

</details>

<details><summary>Logs</summary>

```
# paste relevant logs here, if any
```

</details>

## Screenshots

![bug](https://user-images.githubusercontent.com/129345326/228704247-82cf532f-6d51-487e-8b10-0c4621b6d10c.gif)

## How to reproduce

1. Go to 'Predict'
2. Click on 'Run training...'
3. Select "single animal" for training type
4. Go to "Single Instance Model Configuration" and change any setting
5. Click on "Save configuration files..."
6. Can't save

**Comment by roomrys** [comment]
Hi @bdy2530,

Thanks for filing an issue! I tried this on my SLEAP install (v1.3.0, Windows 10, conda from source) and it is working over here - which isn't very helpful to solving the problem.... I'll switch over to the mac to test there, but am wonderiong if you see any messages in the terminal?

Thanks,
Liezl

**Comment by bdy2530** [comment]
Hi @roomrys,

Thank you for answering! Here is what I see in the terminal after pressing "Save configuration files..."

Traceback  (most recent call last):
  File "/Users/baran/sleap/sleap/gui/learning/dialog.py", line 726, in save
    output_dir = FileDialog.openDir(
  File "/Users/baran/sleap/sleap/gui/dialogs/filedialog.py", line 87, in openDir
    return QtWidgets.QFileDialog.getExistingDirectory(*args, **kwargs)
AttributeError: PySide2.QtWidgets.QFileDialog.getExistingDirectory(): unsupported keyword 'directory'

Best,
Baran

**Comment by Aramist** [comment]
Hi @roomrys,

I just ran into the same issue (Mac OS, Apple silicon, tested both Mamba install and source install from most recent commit on the develop branch) and resolved it by changing the name of the `directory` keyword in [this function call ](https://github.com/talmolab/sleap/blob/1c2be11400481ad2d4f3805010f10e65ff2eb0d3/sleap/gui/learning/dialog.py#L727) to `dir` to match the [documentation for that method](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QFileDialog.html#PySide2.QtWidgets.PySide2.QtWidgets.QFileDialog.getExistingDirectory). I came across this post while checking to see if others have already raised this issue before issuing a PR of my own.

That being said, since you mentioned this function works fine on windows, I'm worried that there might be some platform specific differences in the QT api and that my change might break things there, but I don't have the hardware to test it. I'm honestly not too familiar with the SLEAP code base so I would love to hear your opinion.

**Comment by roomrys** [comment]
Hi @Aramist,

Yes, it is OS specific. SLEAP has an attribute to determine the OS type, but I see that it is only checking for "linux" whereas it should more likely be checking for "win" (or similar) and bundling mac and linux together as the other option:
https://github.com/talmolab/sleap/blob/1c2be11400481ad2d4f3805010f10e65ff2eb0d3/sleap/gui/dialogs/filedialog.py#L90-L104

Then, we make some modifications to the function call based on the OS type, but again we only checked for linux:
https://github.com/talmolab/sleap/blob/1c2be11400481ad2d4f3805010f10e65ff2eb0d3/sleap/gui/dialogs/filedialog.py#L54-L61

**Comment by roomrys** [comment]
This should be addressed in [SLEAP 1.3.2](https://github.com/talmolab/sleap/releases/tag/v1.3.2).
